fabtests/common/shared.c

```diff
@@ -349,6 +350,79 @@ void ft_free_bit_combo(uint64_t *combo)
 	free(combo);
 }
 
+static int ft_alloc_ctx_array(struct ft_context **mr_array)
+{
+	int i, ret;
+	size_t mr_size;
+	uint64_t access = ft_info_to_mr_access(fi);
+	struct ft_context *context;
+	char *default_buf;
+	uint64_t start_key;
+	char **mr_bufs;
+
+	*mr_array = calloc(opts.window_size, sizeof(**mr_array));
+	if (!*mr_array)
+		return -FI_ENOMEM;
+
+	if (*mr_array == tx_ctx_arr) {
```

Comment:
To avoid passing in 
tx_ctx_arr, tx_buf, tx_mr_size, tx_mr_bufs, tx_mr_key for the tx initialization and
rx_ctx_arr, rx_buf, rx_mr_size, rx_mr_bufs, rx_mr_key for the rx initialization
to just check which one we are setting and pick the correct values accordingly. But I can change it to function parameters if you prefer.

Reply:
But you just set *mr_array to the result of calloc()?